### DISCUSSION TITLE
looking to hire imagick specialist

### DISCUSSION THREAD

**Discussion by francisgalarneau** [ImageMagick/ImageMagick#2331] [Help]
hello,

we do have a website that do process multiple images and speed is very slow right now.  
we are hosted on vultr with 2vcore and would upgrade for a better server, mayby one with a graphic cards added and upgrade the code to take this into account would help?
we would like to hire someone that could help us fix that problem.

Please contact at info@pixelpixel.xyz

**Comment by urban-warrior** [comment]
You likely don't need to hire anyone, just post your requirements here.  How many images are processed a day?  What do you do with the images. i.e. what types of image processing algorithms (resize, sharpen, straight conversion)?  What OS are you using?  How much memory is on your system (e.g. 8GB)?  What type of disk are you using?  Have you measured the performance of the disk?  What are the dimensions of the typical image you process (e.g. 1000x1000)?  How many simultaneous users typically are processing images in parallel?  For your web host, how many cores could you request (e.g. 2, 4, 8)?  What is the maximum memory you get request from your web host (e.g. 16GB). And finally, have you read https://imagemagick.org/script/security-policy.php?

**Reply by francisgalarneau** [reply]
well, i have zero progaming knowledge. i do understand the concept and idea behind it, but cant program a single line.  it is an app that has been developp outsourced in asia and i think the developper has reach their maximum knowledge to make it the most efficient. this is why i think i need to hire someone who really know well imagemagick. so mainly its about resizing and merging two image together. one process merge one picture ( could be any size) with 38 other pictures 1000*1000). it takes not about 3minutes 45 secondes. its hosted on vultr has 2vcore 8 gig of ram (will upgrade when site will go online but wonder what is my best option) the cpu seem to never go beyond 15%-20% i did upgrade to a high frequency on vultr and more core but process went only a bit faster (15seconds faster). right now the website is offline... so no other user is using that process, but it could be as 1 or 2 user at the same time. Thank you.

**Comment by urban-warrior** [comment]
We converted 38 1000x1000 images on a 8GB 2 core VM in 6.5 seconds.  Are you using ImageMagick version 6 or 7?  Version 7 defaults to higher precision with floating point which could be a significant performance hit.  You can build ImageMagick 7 and disable HDRI processing.  You can also set a max memory limit in the security policy to ensure all the memory is not consumed and swapping occurs, which can cause system thrashing.  Otherwise, without knowing exactly what your workflow is, we likely cannot help further.
